builder/azure/arm/builder.go

```diff
@@ -74,6 +76,12 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	if b.config.useMSI() {
+		if err := newConfigRetriever().FillParameters(b.config); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
It's kind of ugly to have to copy this code, but I understand why it is necessary now (auth method detection logic). I have a refactor of the azure client config cooking in local repo, I'll try to pick this up then. For now can you add a comment indicating that we shouldn't run this earlier because otherwise ClientConfig will pick the wrong authentication method?

Reply:
Added Comment